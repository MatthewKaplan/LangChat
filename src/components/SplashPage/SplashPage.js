import React, { Component } from 'react';
import { Actions } from 'react-native-router-flux';
import { View, Text, Image, StyleSheet, TouchableOpacity } from 'react-native';
import Button from '../common/Button';

export default class SplashPage extends Component {
	render() {
		const { imageStyle, textStyle, textViewStyle, logInContainerStyle, logInTextStyle, logInButtonStyle } = styles;
		return (
			<View style={imageStyle}>
				<View style={textViewStyle}>
					<Text style={textStyle}>LangChat</Text>
				</View>
				<Image style={imageStyle} source={{ uri: 'https://i.imgur.com/zciZ8K6.jpg' }} />
				<Button onPress={() => Actions.signUpForm()}>Sign Up</Button>
				<View style={logInContainerStyle}>
					<Text style={logInTextStyle}>Already have an account?</Text>
					<Text style={logInButtonStyle} onPress={() => Actions.loginForm()}>
						Log In
					</Text>
				</View>
			</View>
		);
	}
}

const styles = StyleSheet.create({
	textStyle: {
		fontSize: 70,
		fontWeight: '700',
		color: '#fff',
		marginTop: "93%"
	},
	imageStyle: {
		flex: 1,
		alignSelf: 'stretch',
		width: null,
		backgroundColor: "#000",
		opacity: 0.9
	},
	textViewStyle: {
		position: 'absolute',
		zIndex: 1,
		alignItems: 'center',
		justifyContent: 'center',
		paddingTop: 60,
		width: '100%'
	},
	logInContainerStyle: {
		position: 'absolute',
		top: '93%',
		alignSelf: 'center',
		fontWeight: '600',
		flexDirection: 'row'
	},
	logInTextStyle: {
		fontSize: 15,
		color: '#fff'
	},
	logInButtonStyle: {
		fontSize: 15,
		color: '#fff',
		marginLeft: 8,
		textDecorationLine: 'underline'
	}
});
