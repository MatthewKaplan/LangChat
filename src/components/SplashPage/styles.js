import { StyleSheet } from 'react-native';

export default StyleSheet.create({
	textStyle: {
		fontSize: 70,
		fontWeight: 'bold',
		color: '#323232'
	},
	container: {
		flex: 1,
		position: 'relative',
		backgroundColor: '#fff',
		justifyContent: 'center',
		alignContent: 'center'
	},
	imageStyle: {
		flex: 1,
		flexDirection: 'row',
		backgroundColor: '#fff',
	},
	flexContainer: {
		position: 'absolute',
		height: '100%',
		alignItems: 'center',
		justifyContent: 'space-between',
		flexDirection: 'column',
		paddingTop: 75,
		paddingBottom: 75,
		width: '100%',
		alignSelf: 'center'
	},
	logInContainerStyle: {
		justifyContent: 'center',
		fontWeight: '600',
		flexDirection: 'row',
		width: '100%'
	},
	logInTextStyle: {
		fontSize: 16,
		color: '#323232'
	},
	logInButtonStyle: {
		fontSize: 16,
		color: '#323232',
		marginLeft: 8,
		textDecorationLine: 'underline'
	}
});
