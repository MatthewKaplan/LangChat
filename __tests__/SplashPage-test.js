// __tests__/SplashPage-test.js
import React from 'react';
import SplashPage from '../src/components/SplashPage/SplashPage';

import renderer from 'react-test-renderer';

describe('SplashPage Component', () => {
	it('should match snapshot', () => {
		const tree = renderer.create(<SplashPage />).toJSON();
		expect(tree).toMatchSnapshot();
	});
});
