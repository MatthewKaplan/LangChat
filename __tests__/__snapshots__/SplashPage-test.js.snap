// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplashPage Component should match snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#000",
      "flex": 1,
      "position": "relative",
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://images.unsplash.com/photo-1548716188-9b0867f0d1c0?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=934&q=80",
      }
    }
    style={
      Object {
        "backgroundColor": "#000",
        "flex": 1,
        "opacity": 0.75,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "flexDirection": "column",
        "height": "100%",
        "justifyContent": "space-between",
        "paddingBottom": 75,
        "paddingTop": 75,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 70,
          "fontWeight": "bold",
        }
      }
    >
      LangChat
    </Text>
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#007aff",
            "borderRadius": 5,
            "margin": 10,
            "opacity": 1,
            "width": "90%",
          }
        }
      >
        <Text
          style={
            Object {
              "alignSelf": "center",
              "color": "#fff",
              "fontSize": 19,
              "fontWeight": "600",
              "paddingBottom": 12,
              "paddingTop": 12,
            }
          }
        >
          Sign Up
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "fontWeight": "600",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 16,
            }
          }
        >
          Already have an account?
        </Text>
        <Text
          onPress={[Function]}
          style={
            Object {
              "color": "#fff",
              "fontSize": 16,
              "marginLeft": 8,
              "textDecorationLine": "underline",
            }
          }
        >
          Log In
        </Text>
      </View>
    </View>
  </View>
</View>
`;
